x-pack/plugins/actions/server/actions_client.ts

```diff
@@ -298,13 +299,19 @@ export class ActionsClient {
   public async execute({
     actionId,
     params,
+    source,
   }: Omit<ExecuteOptions, 'request'>): Promise<ActionTypeExecutorResult<unknown>> {
-    await this.authorization.ensureAuthorized('execute');
-    return this.actionExecutor.execute({ actionId, params, request: this.request });
+    if (!(await shouldLegacyRbacApplyBySource(this.unsecuredSavedObjectsClient, source))) {
+      await this.authorization.ensureAuthorized('execute');
+    }
```

Comment:
I was hoping to keep this kind of check limited to the Authorization class itself, but as this authorization will have been initialized prior to us *knowing* what the source is, that isn't really possible without adding the `source` as an argument to the `ensureAuthorized` method, which actually felt like a worse solution as most operations won't have that.
Considering we'll be able to remove this all together in 8.0.0 I think it's best to keep this check here as is.